packages/admin-frontend/pages/admin/featuredEdit.js

```diff
@@ -0,0 +1,233 @@
+//@flow
+import React from 'react';
+import { Button, Dialog, TextField, FormHelperText } from '@material-ui/core';
+import { Form, Field, FormSpy } from 'react-final-form';
+import isEmptyString from '../../lib/isEmptyString';
+import UploadFileDialog from '../../components/UploadFileDialog';
+import Row from '../../components/Row';
+import FeaturedImage from '../../components/FeaturedImage';
+import type { FeaturedContent } from '../../types';
+
+type Props = {
+  i: number,
+  button: any,
+  featuredContentList: Array<any>,
+  selectedLanguage: string,
+  defaultReturned: boolean,
+  handleSaveButtonClick: any,
+  handleFileChosen: any,
+  handleOnCancel: any,
+  file: any,
+  handleOnUpload: any
+};
+type State = {
+  open: boolean
+};
+function handleValidate(values) {
+  const errors = {};
+
+  if (isEmptyString(values.title)) {
+    errors.title = 'Required';
+  }
+
+  if (isEmptyString(values.description)) {
+    errors.description = 'Required';
+  }
+
+  const regex = /http(s)?:\/\/.*/;
+  if (isEmptyString(values.link) || !values.link.match(regex)) {
+    errors.link = 'Must be a valid URL e.g "https://digitallibrary.io"';
+  }
+
+  if (isEmptyString(values.imageUrl) || !values.imageUrl.match(regex)) {
+    errors.imageUrl =
+      'Must be a valid URL image url e.g "https://images.digitallibrary.io/imageId.png';
+  }
+
+  return errors;
+}
+
+export default class FeaturedEdit extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      open: false
+    };
+  }
+
+  handleClickOpen() {
+    this.setState({
+      open: true
+    });
+  }
+  handleClose() {
+    this.setState({
+      open: false
+    });
+  }
+
+  handleSaveButtonClick = (defaultReturned: boolean) => (
+    content: FeaturedContent
+  ) => {
+    this.props.handleSaveButtonClick(defaultReturned, content);
+    this.setState({ open: false });
+  };
+  render() {
```

Comment:
Et triks her, hvis du sier (før return):
```javascript
const {button, featuredContentList, selectedLanguage } = this.props;
```
 så slipper du å ha `this.props.selectedLanguage` osv nedover, da kan du bare si `selectedLanguage` :)